### ISSUE TITLE
Trace topology view

### ISSUE THREAD

**Issue by codefromthecrypt** [ymtdzzz/otel-tui#128]
would be nice to be able to aggregate service dependency links and show them

This might use code similar to https://github.com/jaegertracing/jaeger/blob/eaacbf0a40da0c3d55bfeed8d6894f14dca1d4fc/plugin/storage/memory/memory.go#L96

I once wrote one in zipkin, which has more edge cases covered maybe due to duplicate client spans and endpoint change heuristics
https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin2/internal/DependencyLinker.java

Whatever way, I guess the first version doesn't need to be too fancy

**Comment by ymtdzzz** [comment]
Yeah, that's one of the features I want, such like Jaeger's dependency graph:

![image](https://github.com/user-attachments/assets/d869e9f2-c07d-4966-b1eb-a3d449c29a38)
cf. https://signoz.io/blog/jaeger-vs-signoz/

> Whatever way, I guess the first version doesn't need to be too fancy

Right, creating a data model and printing a simple graph (or just text?) from it would be the start.

I'll have a look at it!

**Comment by codefromthecrypt** [comment]
I know the layout will likely use features like you already have in span or detail, but I thought a little about hugo rendering https://gohugo.io/content-management/diagrams/

Also, we probably need to have a zoom. e.g. if only a few services are involved, they will fit on the current columns of the terminal. Otherwise, maybe need to zoom out or allow scrolling with center anchored on last trace viewed (of any). Probably there are other ideas.

cc @tacigar not sure you've seen this it has interesting challenges being a terminal UI!

**Comment by ymtdzzz** [comment]
After some research, I found that [mermaid-ascii](https://github.com/AlexanderGrooff/mermaid-ascii) might be a good choice. I will give it a few tries.

**Comment by codefromthecrypt** [comment]
exciting!

**Comment by ymtdzzz** [comment]
I'm closing this issue because it's implemented in #197 !

![Image](https://github.com/user-attachments/assets/7590a120-a665-4b97-98bb-9b6fca0f4f5f)

However, it is a beta feature, so if you have any issues or ideas for improvement, please let me know!

**Comment by codefromthecrypt** [comment]
thanks a lot and here's the first enhancement request https://github.com/ymtdzzz/otel-tui/issues/200
